File: software/examples/perl/example_scan_for_tags.pl
#!/usr/bin/perl  

use Tinkerforge::IPConnection;
use Tinkerforge::BrickletNFCRFID;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'hjw'; # Change to your UID

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $nfc = Tinkerforge::BrickletNFCRFID->new(&UID, $ipcon); # Create device object
my $tag_type = 0;

# Callback function for state changed callback
sub cb_state_changed
{
	my ($state, $idle) = @_;

	if($idle) {
		$tag_type = ($tag_type + 1) % 3;
		$nfc->request_tag_id($tag_type);
	}

	if($state == $nfc->STATE_REQUEST_TAG_ID_READY) {
		my ($tag_type_new, $tid_length, $tid) = $nfc->get_tag_id();
		my $s = "Found tag of type " . $tag_type . " with ID [" . sprintf("%x", @{$tid}[0]);
		for($i = 1; $i < $tid_length; $i++) {
			$s = $s . ", " . sprintf("%x", @{$tid}[$i]);
		}
		$s = $s . "]\n";

		print $s;
	}
}

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Register state changed callback to function cb_state_changed
$nfc->register_callback($nfc->CALLBACK_STATE_CHANGED, 'cb_state_changed');

$nfc->request_tag_id($nfc->TAG_TYPE_MIFARE_CLASSIC);

print "Press any key to exit...\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/csharp/ExampleScanForTags.cs
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change to your UID

	private static byte currentTagType = BrickletNFCRFID.TAG_TYPE_MIFARE_CLASSIC;

	// Callback function for state changed callback
	static void StateChangedCB(BrickletNFCRFID sender, byte state, bool idle)
	{
		if(idle)
		{
			currentTagType = (byte)((currentTagType + 1) % 3);
			sender.RequestTagID(currentTagType);
		}

		if(state == BrickletNFCRFID.STATE_REQUEST_TAG_ID_READY)
		{
			byte tagType;
			byte tidLength;
			byte[] tid;

			sender.GetTagID(out tagType, out tidLength, out tid);

			if(tidLength == 7)
			{
				System.Console.WriteLine("Found tag of type {0} with ID " +
				                         "[{1:X} {2:X} {3:X} {4:X} {5:X} {6:X} {7:X}]",
				                         tagType, tid[0], tid[1], tid[2], tid[3], tid[4], tid[5], tid[6]);
			}
			else
			{
				System.Console.WriteLine("Found tag of type {0} with ID [{1:X} {2:X} {3:X} {4:X}]",
				                         tagType, tid[0], tid[1], tid[2], tid[3]);
			}
		}
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletNFCRFID nfcrfid = new BrickletNFCRFID(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register state changed callback to function StateChangedCB
		nfcrfid.StateChanged += StateChangedCB;

		nfcrfid.RequestTagID(BrickletNFCRFID.TAG_TYPE_MIFARE_CLASSIC);

		System.Console.WriteLine("Press enter to exit");
		System.Console.ReadLine();
		ipcon.Disconnect();
	}
}
